core/src/main/scala/org/apache/spark/sql/execution/columnar/ExternalStoreUtils.scala

```diff
@@ -275,10 +276,34 @@ object ExternalStoreUtils extends Logging {
     }
     val allPoolProps = getAllPoolProperties(url, driver,
       poolProps, hikariCP, isEmbedded)
-    ConnectionProperties(url, driver, dialect, allPoolProps,
+    getConnectionProperties(session, url, driver, dialect, allPoolProps,
       connProps, executorConnProps, hikariCP)
   }
 
+  def getConnectionProperties(session: Option[SparkSession], url: String, driver: String,
+      dialect: JdbcDialect, poolProps: Map[String, String], connProps: Properties,
+      executorConnProps: Properties, hikariCP: Boolean): ConnectionProperties = {
+    if (session.isDefined && session.get.conf.getOption(Attribute.USERNAME_ATTR).isDefined &&
+        session.get.conf.getOption(Attribute.PASSWORD_ATTR).isDefined) {
```

Comment:
@sumwale This code has been changed since code review. On SNAP-1656, corresponding code is:

`  def getConnectionProperties(session: Option[SparkSession], url: String, driver: String,
      dialect: JdbcDialect, poolProps: Map[String, String], connProps: Properties,
      executorConnProps: Properties, hikariCP: Boolean): ConnectionProperties = {
    if (session.isDefined) {
      val (user, password) = getCredentials(session)

      if (!user.isEmpty && !password.isEmpty) {
        def secureProps(props: Properties): Properties = {
          props.setProperty(Attribute.USERNAME_ATTR, user)
          props.setProperty(Attribute.PASSWORD_ATTR, password)
          props
        }

        // Hikari only take 'username'. So does Tomcat
        def securePoolProps(props: Map[String, String]): Map[String, String] = {
          props + (Attribute.USERNAME_ALT_ATTR.toLowerCase -> user) + (Attribute.PASSWORD_ATTR ->
              password)
        }

        ConnectionProperties(url, driver, dialect, securePoolProps(poolProps),
          secureProps(connProps), secureProps(executorConnProps), hikariCP)
      } else {
        ConnectionProperties(url, driver, dialect, poolProps, connProps, executorConnProps,
          hikariCP)
      }
    } else ConnectionProperties(url, driver, dialect, poolProps, connProps, executorConnProps,
      hikariCP)
  }

  def getCredentials(session: Option[SparkSession], prefix: String = ""): (String, String) = {
    val prefix = SnappyContext.getClusterMode(session.get.sparkContext) match {
      case ThinClientConnectorMode(_, _) => Constant.SPARK_STORE_PREFIX
      case _ => ""
    }
    (session.get.conf.get(prefix + Attribute.USERNAME_ATTR, ""),
        session.get.conf.get(prefix + Attribute.PASSWORD_ATTR, ""))
  }`